Rx.NET/Documentation/adr/0001-multi-platform-packaging.md

```diff
@@ -0,0 +1,106 @@
+
+* In v1, Microsoft built multiple different versions of the Rx libraries for different platforms. (This was long before .NET Standard or even before Portable Class Libraries.)
+* v2 was designed to take advantage of the (then new, now defunct) Portable Class Library (PCL) concept. Assemblies were in one of these categories:
+  * A portable core
+    * `System.Reactive.Interfaces` (in the [`Rx-Interfaces`](https://www.nuget.org/packages/Rx-Interfaces/2.2.5) NuGet package)—originally conceived as an assembly that would stay on v2.0.0.0 indefinitely, containing the canonical definitions of core interfaces
+    * `System.Reactive.Core` (in the [`Rx-Core`](https://www.nuget.org/packages/Rx-Core/2.2.5) NuGet package)—platform-independent schedulers, and utility types for implementing Rx operators
+    * `System.Reactive.Linq` (in the [`Rx-Linq`](https://www.nuget.org/packages/Rx-Linq/2.2.5) NuGet package)—implementations of LINQ operators for Rx
+    * The [v2.0 beta announcement](http://web.archive.org/web/20130522225545/http://blogs.msdn.com/b/rxteam/archive/2012/03/12/reactive-extensions-v2-0-beta-available-now.aspx) mentions `System.Reactive.Providers`—expression tree support for Rx LINQ—but I don't see this on NuGet, so I'm not sure if it got rolled into something else
+  * Platform-specific services, just one component, `System.Reactive.PlatformServices`, but the `Rx-PlatformServices` package contained builds for multiple target platforms including:
+    * .NET Framework 4.0
+    * .NET Framework 4.5
+    * Silverlight 5
+    * Windows Phone (7, 7.1, and 8)
+    * PCL (various profiles, covering .NET Framework, Silverlight, and several versions of Windows Phone)
+  * UI-framework-specific functionality
+    * `System.Reactive.Windows.Threading` (in the [`Rx-Xaml`](https://www.nuget.org/packages/Rx-Xaml/2.2.5) NuGet package)—(WPF and WinRT schedulers and extension methods)
+    * `System.Reactive.Windows.Forms` (in the [`Rx-WinForms`](https://www.nuget.org/packages/Rx-WinForms/2.2.5) NuGet package)—(Windows Forms schedulers and extension methods)
+    * `System.Reactive.WindowsRuntime` (in the [`Rx-Metro`](https://www.nuget.org/packages/Rx-Metro/2.2.5) NuGet package)—(conversions between WinRT types and)
+* v3 changed the NuGet package names, but essentially kept the same structure
+  * The old `Rx-Main` metapackage is replaced by `System.Reactive`
+  * `Rx-Interfaces` becomes `System.Reactive.Interfaces` (v3.0, not clear if there were actually interface changes)
+  * `Rx-Core` becomes `System.Reactive.Core`
+  * `Rx-Linq` becomes `System.Reactive.Linq`
+  * `Rx-PlatformServices` becomes `System.Reactive.PlatformServices`
+  * Likewise, the framework-specific libraries `System.Reactive.Windows.Threading`, `System.Reactive.Windows.Forms`, and `System.Reactive.WindowsRuntime` move into eponymous packages
+* v4 made a major change: everything is now in a single `System.Reactive` package
+* v5 stuck with the same structure as v4
```

Comment:
.NET Framework 4.7.2 was selected as the lowest version for it's .NET Standard 2.0 compatibility and the requirement to be able to use ValueTuple.